lib/types/string/index.js

```diff
@@ -315,7 +315,14 @@ internals.String = class extends Any {
         return this._test('isoDate', undefined, function (value, state, options) {
 
             if (JoiDate._isIsoDate(value)) {
-                return value;
+                if (!options.convert) {
+                    return value;
+                }
+
+                try {
+                    return new Date(value).toISOString();
+                }
+                catch (err) { }
```

Comment:
I now return the isoDate Error in the catch block with some additional details (the thrown err which will say why it was unable to convert to ISO string, and the fact that convert is true) instead of just falling through to the standard isoDate error.  Would you prefer I add a new error type (i.e. isoDateConvert) that has a message that explicitly calls out that it was the conversion that failed?